projectview/src/main/java/org/jetbrains/bsp/bazel/projectview/parser/splitter/ProjectViewSectionSplitter.java

```diff
@@ -1,91 +1,52 @@
 package org.jetbrains.bsp.bazel.projectview.parser.splitter;
 
+import com.google.common.collect.Streams;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.ListIterator;
-import java.util.Optional;
-import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public final class ProjectViewSectionSplitter {
 
-  private static final String SECTION_HEADER_REGEX = "(^[a-z_]+[:]?)";
-  private static final String POSSIBLE_HEADER_ENDING_CHARACTER = ":";
+  private static final Pattern SECTION_HEADER_REGEX =
+      Pattern.compile("((^[^:\\-/*\\s]+)([: ]))", Pattern.MULTILINE);
+  private static final int SECTION_HEADER_NAME_GROUP_ID = 2;
 
-  public static List<ProjectViewRawSection> split(String fileContent) {
-    List<ProjectViewSectionHeaderPosition> sectionsHeadersPositions =
-        splitIntoSectionsHeadersPositions(fileContent);
+  private static final String COMMENT_LINE_REGEX = "#(.)*(\\n|\\z)";
+  private static final String COMMENT_LINE_REPLACEMENT = "\n";
 
-    return parseRawSections(fileContent, sectionsHeadersPositions);
-  }
+  public static ProjectViewRawSections split(String fileContent) {
+    var fileContentWithoutComments = removeLinesWithComments(fileContent);
```

Comment:
No var. Same for lines 21, 31, 32, 39 and 40.

Reply:
why?

Reply:
So it's clear what the types are. Probably not an issue for lines 21 and 39 though, since their names give away (something clear) about the types.

Reply:
well, IntelliJ says that in a very pretty way